File: src/lightnap-ng/src/app/core/backend-api/static-content-read-access.ts
import { ListItem } from "@core/models/list-item";
import { StaticContentReadAccess } from "./models";

export const StaticContentReadAccessListItems = [
  new ListItem<StaticContentReadAccess>(StaticContentReadAccess.Public, "Public", "Visible to anyone."),
  new ListItem<StaticContentReadAccess>(StaticContentReadAccess.Authenticated, "Authenticated", "Visible to authenticated users."),
  new ListItem<StaticContentReadAccess>(StaticContentReadAccess.Explicit, "Explicit", "Visible to explicitly granted roles and users."),
];

File: src/LightNap.Core/StaticContents/Enums/StaticContentReadAccess.cs
namespace LightNap.Core.StaticContents.Enums
{
    /// <summary>
    /// The read access of a CMS item.
    /// </summary>
    public enum StaticContentReadAccess
    {
        /// <summary>
        /// Anyone may request the content.
        /// </summary>
        Public = 0,

        /// <summary>
        /// Users must be logged in to request the content.
        /// </summary>
        Authenticated = 1,

        /// <summary>
        /// The content is only visible to specified roles and users.
        /// </summary>
        Explicit = 2,
    }
}
